src/net.cpp

```diff
@@ -432,7 +432,8 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         std::string host;
         int port = default_port;
         SplitHostPort(std::string(pszDest), port, host);
-        connected = ConnectThroughProxy(proxy, host, port, hSocket, nConnectTimeout, nullptr);
+        bool proxyConnectionFailed;
+        connected = ConnectThroughProxy(proxy, host, port, hSocket, nConnectTimeout, proxyConnectionFailed);
```

Comment:
It doesn't especially relevant to this PR so this should block this PR but I noticed that `proxyConnectionFailed` is not checked in this call Does it make sense to create a ConnectThroughProxy method without the `outProxyConnectionFailed` parameter or should this call check `proxyConnectionFailed` like the prior call does?

Reply:
Good point! Let's we tackle that in a separate follow-up PR to keep this one focused :)

Reply:
Agreed, I just wanted to make a note here so I didn't forget.